src/daemon/default_vm_image_vault.cpp

```diff
@@ -281,6 +281,12 @@ mp::VMImage mp::DefaultVMImageVault::fetch_image(const FetchType& fetch_type,
         QUrl image_url(QString::fromStdString(query.release));
         VMImage source_image, vm_image;
 
+        if (image_url.host().size() != 0)
+            throw std::runtime_error(fmt::format("Unexpected hostname in file URL; did you mean "
```

Comment:
I don't think the "unexpected hostname" part would be valuable to the user. Could we say something like, "Malformed file URL, did you miss a slash?" without going into details?